### PR TITLE
8 implement pagination for jobs table

### LINKED ISSUE #8: Implement Pagination for Jobs Table
Currently, all jobs are displayed in a single table, which can become unwieldy with many jobs. Add pagination to improve performance and usability.
- [x] Add page navigation controls (prev/next/page numbers)
- [x] Allow configurable number of jobs per page (e.g., 10, 20, 50)
- [x] Preserve filters when navigating between pages

### PR DESCRIPTION
**Pull request by Anish-Chanda**
fixes #8 

Copilot summary (coz I don't want to type this out)
This pull request introduces pagination functionality to the job listing application, enhances the user interface with pagination controls, and improves code readability and maintainability. The changes include updates to the backend to support pagination, integration of pagination into the frontend, and the addition of tests to verify the new functionality.

### Backend Enhancements:
* Added pagination support in the `getSlurmJobs` function, including calculation of page size, total items, and total pages. The function now slices job results based on the specified page and page size. (`handlers/fetchJobs.js`, [handlers/fetchJobs.jsL14-R64](diffhunk://#diff-be03c0d258c20fc2e0fa2b26240704c8923a0e65419ce120f0548ac11dd2bd20L14-R64))
* Introduced a new constant `DEFAULT_PAGE_SIZE` to standardize the default page size across the application. (`constants.js`, [constants.jsR1](diffhunk://#diff-e7ad5f13058c7d8fe78b1910deadef9c6f714595824f73d12f572e8b7e9d258dR1))

### Frontend Updates:
* Integrated pagination controls into the `home.handlebars` view, including a dropdown to select jobs per page and buttons for navigating between pages. (`views/home.handlebars`, [[1]](diffhunk://#diff-dbed998603a94ea423636bb9aa34e71e6253bb771a3d7b51311f17a741294c1eR66-R121) [[2]](diffhunk://#diff-dbed998603a94ea423636bb9aa34e71e6253bb771a3d7b51311f17a741294c1eR329-R393)
* Updated the `fetchJobs` function to handle pagination parameters and dynamically update the UI based on the current page and page size. (`views/home.handlebars`, [[1]](diffhunk://#diff-dbed998603a94ea423636bb9aa34e71e6253bb771a3d7b51311f17a741294c1eL206-R272) [[2]](diffhunk://#diff-dbed998603a94ea423636bb9aa34e71e6253bb771a3d7b51311f17a741294c1eR287)
* Added event listeners for pagination buttons and page size selectors to ensure proper interaction with the server-rendered and dynamically updated views. (`views/home.handlebars`, [views/home.handlebarsR436-R462](diffhunk://#diff-dbed998603a94ea423636bb9aa34e71e6253bb771a3d7b51311f17a741294c1eR436-R462))

### Testing Improvements:
* Added comprehensive tests for the `getSlurmJobs` function to verify pagination behavior, including handling of custom page sizes, filters, and edge cases like out-of-range pages. (`tests/handlers/fetchJobs.test.js`, [tests/handlers/fetchJobs.test.jsR131-R224](diffhunk://#diff-830b299de7c948660a40ce3649a831e515a00860b2fab1b4424cdf18be65b1bfR131-R224))

### Code Quality Improvements:
* Refactored code for consistency, such as replacing single quotes with double quotes in `handlers/fetchJobs.js` for uniformity. (`handlers/fetchJobs.js`, [handlers/fetchJobs.jsL14-R64](diffhunk://#diff-be03c0d258c20fc2e0fa2b26240704c8923a0e65419ce120f0548ac11dd2bd20L14-R64))
* Enhanced the Handlebars engine configuration by adding custom helpers for mathematical operations and pagination logic. (`app.js`, [app.jsR5-R63](diffhunk://#diff-e07d531ac040ce3f40e0ce632ac2a059d7cd60f20e61f78268ac3be015b3b28fR5-R63))

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,22 +2,65 @@ const express = require('express');
...
 
 router.get('/api/jobs', async (req, res) => {
-  const result = getSlurmJobs(req.query);
-  res.json(result)
+  const { page, pageSize, ...filters } = req.query;
+
+  const pagination = {
+    page: page ? parseInt(page) : 1,
+    pageSize: pageSize ? parseInt(pageSize) : DEFAULT_PAGE_SIZE
+  }
```
[the remark this answers is not in the document]
Comment (Anish-Chanda) [reply app.js:60]: If the page param is NaN, we will fall back to page 1 already. IMO this is unnecessary.
